java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java

```diff
@@ -3509,9 +3509,9 @@ void testCastFloatToDecimal() {
   @Test
   void testCastDoubleToDecimal() {
     testCastNumericToDecimalsAndBack(DType.FLOAT64, false, 0,
-        () -> ColumnVector.fromBoxedDoubles(1.0, 2.1, -3.23, null, 2.41281, (double) Long.MAX_VALUE),
-        () -> ColumnVector.fromBoxedDoubles(1.0, 2.0, -3.0, null, 2.0, (double) Long.MAX_VALUE),
-        new Long[]{1L, 2L, -3L, null, 2L, Long.MAX_VALUE}
+        () -> ColumnVector.fromBoxedDoubles(1.0, 2.1, -3.23, null, 2.41281, (double) Integer.MAX_VALUE),
+        () -> ColumnVector.fromBoxedDoubles(1.0, 2.0, -3.0, null, 2.0, (double) Integer.MAX_VALUE),
+        new Long[]{1L, 2L, -3L, null, 2L, Integer.MAX_VALUE.longValue()}
```

Comment:
Curious why the values were changed.  This test used to pass, so I assume that we no longer support round-trip casting of Long.MAX_VALUE to double and back like we used to.  I can see how that could be problematic.

Nit: While we're changing values, it would be nice to test at least _something_ that's more interesting and higher precision than what can be trivially stored in a 32-bit mantissa given we've got another 20 bits to play with there.  Long.MAX_VALUE was definitely leveraging those extra bits, but integer.MAX_VALUE is not.

Reply:
The problem with this conversion is that long.MAX_VALUE contains 63 bits of precision (all 1s), more than can be stored in double (52).  So in converting that to double, it gets stored in the double as 2^64 (the closest value of double).  Then the conversion algorithm faithfully converts that double back to an unsigned long long of 2^64, and since dec64 is signed, it overflows on the final output cast.

Reply:
The best way to handle this case would be to have the code detect overflows and handle them appropriately.  It's not immediately clear what that should be (set to max? to null?), and detecting overflow was not part of the old algorithm either. I've commented in the code where overflow can occur, so that when we decide how we want to handle that it in a future PR.